calendar_export_ics/wizards/wizard_export_ics.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (C) 2024 - ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+import base64
+from datetime import date
+
+from odoo import api, fields, models
+
+
+class CalendarExportIcs(models.TransientModel):
+    """
+    This wizard is used to export odoo calendar to ics file
+    """
+
+    _name = "calendar.export.ics"
+    _description = "Calendar Export Ics"
+
+    export_ics_file = fields.Binary("Ics Exported File")
+    export_ics_filename = fields.Char(readonly=True)
+    export_end_date = fields.Date("End Export Date")
+    partner_id = fields.Many2one("res.partner", string="Partner")
+
+    @api.model
+    def default_get(self, fields):
+        defaults = super(CalendarExportIcs, self).default_get(fields)
+        user = self.env.user
+        if user.partner_id:
+            defaults["partner_id"] = user.partner_id.id
+        return defaults
+
+    def button_export(self):
+        today_date = date.today()
+        self.export_ics_filename = today_date.strftime("%Y-%m-%d") + "_calendar.ics"
+        self.export_ics_file = self.generate_ics_content()
+
+        return {
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "res_model": "calendar.export.ics",
+            "name": "Export to Ics File",
+            "target": "new",
+            "res_id": self.id,
+        }
+
+    def generate_ics_content(self):
```

Comment:
This function seems a bit DRY with respect to odoo/odoo `calendar.event` _get_ics_file
https://github.com/odoo/odoo/blob/f219583052e46ae8942675495c3833bd432dacd8/addons/calendar/models/calendar_event.py#L1056-L1114. Given that it does not make use of the `vobject` package, are there any concerns that this introduces .ics syntax errors or attack surfaces for injection?

Reply:
See https://github.com/odoo/odoo/pull/166282 for an attempt to avoid DRY.

Reply:
Hi @amh-mw,
You are right, it looks a bit DRY, i didn't know about the function _get_ics_file. 
Now the code uses this function to avoid DRY